### ISSUE TITLE
Cannot get 

### ISSUE THREAD

**Issue by AndrewJDFerguson** [archetyped/simple-lightbox#972]
## Description of Problem
I tried to get this to work on a site using the Twenty Twenty-Two theme. Simple gallery containing six images and set to link the images to the media file.

Clicking on an image just shows the image, no controls to switch between images.
![SLB-Issue](https://user-images.githubusercontent.com/37160315/195626133-0eeb4f38-77d1-406e-9aaa-10bdf59a9fc7.png)

## Details

* URL of Page exhibiting problem:  This is currently on a local site and not accessible from the internet. I will not put this on a live site at this time.
* Browser(s) exhibiting problem (Firefox, Chrome, etc.): Edge (up date)
* Version of SLB Installed: 2.9.2
* Other plugins installed: Gutenberg (latest), Simple Light Box, SQL Buddy

**Comment by archetyped** [comment]
Hi, a link to a page that exhibits the specific issue you are experiencing is the fastest way to determine the root cause. Without a link to evaluate the issue directly, following these [steps for evaluating plugin issues](https://github.com/archetyped/simple-lightbox/wiki/Reporting-Issues#first-steps) is recommended.

Please provide the details collected from those steps (e.g. PHP/MySQL versions, cURL configuration, etc.), as well as the method you are using for running the local site, and I would be glad to take a look.

Thanks.

**Comment by archetyped** [comment]
A few more details to confirm:

* You didn't mention the [jQuery version](https://github.com/archetyped/simple-lightbox/wiki/Requirements#wordpress) loaded on the frontend pages.
* Is the local site's URL set to use `https://`?  This can create issues especially on local sites if the configuration/certificates are not set up properly.
* Are there any errors/messages in the browser's developer console for frontend pages?  Even error messages from other plugins may be important because an error can stop other scripts from functioning normally.
* Do the image links on the page have SLB's custom attributes (e.g. `data-slb-active`)?
* Please provide a screenshot of SLB's options page.

**Comment by AndrewJDFerguson** [comment]
- You didn't mention the [jQuery version](https://github.com/archetyped/simple-lightbox/wiki/Requirements#wordpress) loaded on the frontend pages.

No other plugins are pulling in jQuery, and in looking at the page source, I do not see jQuery being loaded, so have to assume that I am using the version of jQuery loaded with WordPress 6.0.x. 

- Is the local site's URL set to use https://? This can create issues especially on local sites if the configuration/certificates are not set up properly.

The site is set to use https (local self-signed certificate), and the browser is showing that the site is secure and everything is fine)

- Are there any errors/messages in the browser's developer console for frontend pages? Even error messages from other plugins may be important because an error can stop other scripts from functioning normally.

By front-end page, assuming you mean the front end page hosting the Gallery/Images. The page hosting the gallery is showing the following: 

> [Intervention] Images loaded lazily and replaced with placeholders. Load events are deferred. See https://go.microsoft.com/fwlink/?linkid=2048113
> jquery-migrate.js?ver=3.3.2:69 JQMIGRATE: Migrate is installed with logging active, version 3.3.2
> jquery-migrate.js?ver=3.3.2:100 
>         
>        JQMIGRATE: jQuery.type is deprecated
> migrateWarn @ jquery-migrate.js?ver=3.3.2:100
> jquery-migrate.js?ver=3.3.2:102 console.trace
> migrateWarn @ jquery-migrate.js?ver=3.3.2:102
> 

The page loaded when clicking on an image shows no console messages.

- Do the image links on the page have SLB's custom attributes (e.g. data-slb-active)?

I see the data-slb-active attribute. Below is the content from the browser:

> <figure class="wp-block-image size-large"><a href="https://wordpress-development.canterbury.local/wp-content/uploads/2022/09/Space-Rocks-1.jpg" data-slb-active="1" data-slb-asset="876103556" data-slb-internal="0" data-slb-group="3666"><img loading="lazy" width="1024" height="819" data-id="3651" src="https://wordpress-development.canterbury.local/wp-content/uploads/2022/09/Space-Rocks-1-1024x819.jpg" alt="" class="wp-image-3651" srcset="https://wordpress-development.canterbury.local/wp-content/uploads/2022/09/Space-Rocks-1-1024x819.jpg 1024w, https://wordpress-development.canterbury.local/wp-content/uploads/2022/09/Space-Rocks-1-300x240.jpg 300w, https://wordpress-development.canterbury.local/wp-content/uploads/2022/09/Space-Rocks-1-768x614.jpg 768w, https://wordpress-development.canterbury.local/wp-content/uploads/2022/09/Space-Rocks-1.jpg 1200w" sizes="(max-width: 1024px) 100vw, 1024px"></a><figcaption class="wp-element-caption">Dusty remains of shredded asteroids around several dead stars. Original from NASA. Digitally enhanced by rawpixel.</figcaption></figure>

- Please provide a screenshot of SLB's options page.

Only default options are in used. See attached.
![SLB-Settings](https://user-images.githubusercontent.com/37160315/197003378-c69d45dc-152e-45f9-87e9-a480397c59c9.png)

Thanks

**Comment by archetyped** [comment]
I would love to help, but the assumptions are stacking up.  Each unverified assumption is a potential point of failure.  Without being able to evaluate the issue directly, there's not much I can do from my end things are left unverified.

For example, my previous question regarding jQuery version links to a bookmarklet that makes it very straightforward to verify the version of jQuery being loaded on the page.  Once the version is known, we can determine whether that is contributing to the issue you are experiencing or not.

I also would recommend temporarily setting the site to load via http to see if that is a factor. While the browser may report that things are working, server-side code such as cURL may be encountering issues.

Enabling [WordPress' debug logging](https://wordpress.org/support/article/debugging-in-wordpress/) is also recommended.  Load a page that SLB should be working on, and then review WordPress' log as well as your XAMPP's server logs to see if there are any error messages.

**Note:** the messages may not appear to be directly related to SLB, as an error in one plugin can prevent other plugin's from running, so any error/warning/notice may be important.

Let me know what you find.

**Comment by AndrewJDFerguson** [comment]
Well, since from my experience, there is sometimes a lot of content that gets written to the site content in the database, I am reluctant to switch that site to http from https. That said, I will just spin up a new site using just http and get see if that helps.

Perhaps I missed the jQuery version links. Not really sure. I will get that from the new site I spin up. That said, whatever version of jQuery will be the same that I have seen, as this will be a WP 6.0.x site, like my other.

I have debug logging enabled on the site, and will also make sure it is enabled on the new site. 

I will let you know once I have a chance to do that this week.

**Comment by AndrewJDFerguson** [comment]
So, I spun up a new local site that did not use HTTPS. Same result. I did see an error on the browser console related to a browser extension (FWIW, I did not get this error previously). Based on that I disabled my browser extensions, and the SLB plugin worked on the non-HTTPS site. Interesting. I then went and switched the site back to HTTPS and SLB no longer worked. So, it appears that I have two issues that get in the way of using this plugin in a development/staging setup. Not helpful for me, as I will not deploy this to a production site unless I have it working in development/staging.

That's a deal breaker for me, and my confidence in being able to reliably use this plugin now and going forward is doubtful. Too much effort for something that may or may not work in production, and could easily be "broken" by a browser extension.

I appreciate your time in helping me try to get this to work.

Thanks
